components/scream/src/physics/shoc/shoc_compute_diag_third_shoc_moment_impl.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef SHOC_COMPUTE_DIAG_THIRD_SHOC_MOMENT_IMPL_HPP
+#define SHOC_COMPUTE_DIAG_THIRD_SHOC_MOMENT_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::compute_diag_third_shoc_moment(
+  const MemberType& team,
+  const Int& nlev,
+  const Int& nlevi,
+  const uview_1d<const Spack>& w_sec,
+  const uview_1d<const Spack>& thl_sec,
+  const uview_1d<const Spack>& wthl_sec,
+  const uview_1d<const Spack>& tke,
+  const uview_1d<const Spack>& dz_zt,
+  const uview_1d<const Spack>& dz_zi,
+  const uview_1d<const Spack>& isotropy_zi,
+  const uview_1d<const Spack>& brunt_zi,
+  const uview_1d<const Spack>& w_sec_zi,
+  const uview_1d<const Spack>& thetal_zi,
+  const uview_1d<Spack>& w3)
+{
+  const Int nlev_pack = ekat::pack::npack<Spack>(nlev);
+  const Int nlevi_pack = ekat::pack::npack<Spack>(nlevi);
+
+  // Scalarize views for shifts
+  const auto s_dz_zt = scalarize(dz_zt);
+  const auto s_wthl_sec = scalarize(wthl_sec);
+  const auto s_thl_sec = scalarize(thl_sec);
+  const auto s_w_sec = scalarize(w_sec);
+  const auto s_tke = scalarize(tke);
+
+  // Set lower condition: w3(i,nlevi) = 0
+  const Int last_pack_entry = (nlevi%Spack::n == 0 ? Spack::n-1 : nlevi%Spack::n-1);
+  w3(nlevi_pack-1)[last_pack_entry] = 0.0;
```

Comment:
Always prefer integer literals if you're dealing with a round number so the compiler can promote it to the appropriate type. I see a lot of `X.0` literals in the code below. I think all of them can be `X`

Reply:
Ok, good to know! I've made the change.